dev_guide/builds.adoc

```diff
@@ -846,20 +846,28 @@ source:
 
 == Binary Source
 
-When the `*BuildConfig.spec.source.type*` is `*Binary*`, the build will expect a
-binary as input, and an inline Dockerfile is optional.
-
-The binary is generally assumed to be a tar, gzipped tar, or zip file depending
-on the strategy. For `*Docker*` builds, this is the build context and an
-optional Dockerfile may be specified to override any Dockerfile in the build
-context. For `*Source*` builds, this is assumed to be an archive as described
-above. For `*Source*` and `*Docker*` builds, if `*binary.asFile*` is set the
-build context will consist of a single file named by the value of `*binary.asFile*`. 
-The `*contextDir*` field may be used when an archive is provided. Custom builds will 
-receive this binary as input on standard input (`stdin`).
-
-A binary source potentially extracts content (if `*asFile*` is not set), in which case `*contextDir*`
-allows changing to a subdirectory within the content before the build executes.
+When the `*BuildConfig.spec.source.type*` is `*Binary*`, the source content will be sent
+as input to the build via a binary stream and  `oc start-build` should be executed with either
+the `--from-dir`, `--from-file`, or `--from-repo` options.
+
+The binary stream will be processed in one of two ways:
+
+1) If `*binary.asFile*` is set, then the bytes from the binary stream will be considered a single file.  The binary
+stream will be stored in a file on the builder's file system with the name as specified as `*binary.asFile*` for the builder to use. 
```

Comment:
Yeah, I just included any of the `oc start-build` options that mentioned binary input.  Here is a copy paste for those three from the `oc help start-build` output:

`      --from-dir='': A directory to archive and use as the binary input for a build.`
`      --from-file='': A file to use as the binary input for the build; example a pom.xml or Dockerfile. Will be the only file in the build source.`
`      --from-repo='': The path to a local source code repository to use as the binary input for a build.`

I assumed (and then confirmed by looking at https://github.com/openshift/origin/blob/master/pkg/cmd/cli/cmd/startbuild.go#L214 after seeing your question) that the from-dir and from-repo translate into the client sending an archive of those over the binary stream, and yes, asFile only makes sense with --from-file (though I could see --from-file not needing asFile if say the one file was a pom.xml or Dockerfile).

That sound right to everybody?